src/validations/rules/sar.sch

```diff
@@ -176,6 +175,140 @@
+            <sch:let
+                name="sap-available"
+                value="
+                    if (: this is not a relative reference :) (not(starts-with($sap-import-url, '#')))
+                    then
+                        (: the referenced document must be available :)
+                        doc-available($sap-import-url)
+                    else
+                        true()" />
+            <sch:let
+                name="sap-doc"
+                value="
+                    if ($sap-available)
+                    then
+                        doc($sap-import-url)
+                    else
+                        ()" />
+            <sch:let
+                name="ssp-import-url"
+                value="resolve-uri($sap-doc/oscal:assessment-plan/oscal:import-ssp/@href, base-uri())" />
+            <sch:let
+                name="ssp-available"
+                value="
+                    if (: this is not a relative reference :) (not(starts-with($ssp-import-url, '#')))
+                    then
+                        (: the referenced document must be available :)
+                        doc-available($ssp-import-url)
+                    else
+                        true()" />
+            <sch:let
+                name="ssp-doc"
+                value="
+                    if ($ssp-available)
+                    then
+                        doc($ssp-import-url)
+                    else
+                        ()" />
+            <sch:let
+                name="profile-import-url"
+                value="resolve-uri($ssp-doc//oscal:import-profile/@href, base-uri())" />
+            <sch:let
+                name="profile-available"
+                value="
+                    if (: this is not a relative reference :) (not(starts-with($profile-import-url, '#')))
+                    then
+                        (: the referenced document must be available :)
+                        doc-available($profile-import-url)
+                    else
+                        true()" />
+            <sch:let
+                name="profile-doc"
+                value="
+                    if ($profile-available)
+                    then
+                        doc($profile-import-url)
+                    else
+                        ()" />
+            <!-- UUID of most recent result -->
+            <sch:let
+                name="first-result"
+                value="//oscal:result[xs:dateTime(oscal:start) eq max(//oscal:result/oscal:start ! xs:dateTime(.))]/@uuid" />
+            <!-- In-scope controls -->
+            <sch:let
+                name="in-scope-controls"
+                value="$ssp-doc//oscal:implemented-requirement/@control-id" />
+            <!-- Baseline controls -->
+            <sch:let
+                name="baseline-controls"
+                value="$profile-doc//oscal:alter/@control-id" />
```

Comment:
This might work if only FedRAMP profiles were used (and may or may not work whenever FedRAMP provides 800-53rev5 content). There is a complication that the chain from SAR to SAP to SSP (and possibly farther) may or may not have a profile or a resolved profile (i.e., a catalog).

When the term _baseline_ is used, it is a set of controls (and in OSCAL is defined in a profile). _Resolved profiles_ complicate this: the baseline control set might be in a profile (or worse, more than one)or might be in a resolved catalog. 

A profile might not use `alter` - FedRAMP v4 profiles do and have equal numbers of `with-id` and `alter` but we have not yet seen the rev5 variants.